### PR TITLE
[Profile] Other profile screen UI

### LINKED ISSUE #8071: [Profile] Other profile screen UI
image

image

crossed element not in the scope of this PR

Acceptance criteria:

1) pixel perfect
2) new design introduced
3) Profile details introduced
4) prompts introduced

Figma:
https://www.figma.com/file/TNCyHKtR3sx5EL6YznFWUa4O/Profile?node-id=510%3A3116

### PR DESCRIPTION
**Pull request by tbenr**
fixes: #8071

after #8069 as been merged I integrate my part with the work from the great @bitsikka 

regarding large_toolbar:
- I "componentized" more `large_toolbar`, decoupling styles too.
- i changed `large_toolbar`implementation on flat-list's on-scroll events
- shadow animations are now interpolated with scroll
- minimized toolbar fade-in\fade-out is still triggered on a certain point (not interpolated. this can be changed if needed)
- animation variables are now local to the views (if multiple screens uses the component, variables remain separated and state remains consistent while navigating)

then i implemented the "other profile". It is still not supporting multiple details (main chat account is listed) but the basis for listing multiple accounts\wallets is there.

status: ready

### CODE REVIEW

**Comment by flexsurfer** [comment]
@tbenr is this still wip?

**Comment by tbenr** [comment]
@flexsurfer yes, need to remove unused files and a couple of other things.. this evening i should finish.

**Comment by tbenr** [comment]
@flexsurfer I think we can start the review.

File: src/status_im/contact/core.cljs
Comment (flexsurfer) [inline src/status_im/contact/core.cljs]: i believe we don't use `:data-store/tx` for contacts anymore @cammellos could you assist, please?

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/contact/views.cljs]: why is this commented?

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/screens/profile/contact/views.cljs]: could you elaborate?

Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs]: @flexsurfer the function should "prepare" a list of details (accounts\wallet) from the current contact. Since i don't know how will actually wallet\accounts be stored in db for a given contact, I prepared a function for the purpose.
Currently it just create an array with current account.

**Comment by tbenr** [comment]
@flexsurfer i added `:share` accessory to `list-item` component. But I have an issue on `list-item` which I'm struggling to fix:

wrong (long name):
image

ok (short name):
image

the accessory goes off-screen. Any hint?

**Comment by tbenr** [comment]
I also implemented sheet for "block this user"
image

File: src/status_im/ui/components/list_item/views.cljs
Comment (yenda) [inline src/status_im/ui/components/list_item/views.cljs]: already imported under the alias "icons", feel free to change it to "vector-icons" though

**Comment by bitsikka** [comment]
> @flexsurfer i added `:share` accessory to `list-item` component. But I have an issue on `list-item` which I'm struggling to fix:
> 
> wrong (long name):
> image
> 
> ok (short name):
> image
> 
> the accessory goes off-screen. Any hint?

@tbenr let me chime in here.

As a critical part of #8070 I am adjusting the `list-item` namespace as well, and this is related.

- In the structural improvements I have made,
  > wrong (long name):

  does not happen

- Also, the `:share` accessory could be categorized as interactive accessory like `check-box`, `radio-button` or `toggle-switch` for which the `:accessories` vector already accepts `component` as an accessory. So, instead of adding `:share` accessory in the `list-item` namespace, you could pass the `share button` as a component in `:accessories` vector. cc @flexsurfer @errorists 

I am almost done with the PR for #8070 and will be pushing it soon

**Comment by bitsikka** [comment]
btw @tbenr 
- cool alternative implementation of `large-toolbar` 👍
- thank you for extracting the styles in separate stylesheet too 🙏

**Comment by tbenr** [comment]
@bitsikka thanks for sharing!

from https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=794%3A173

image

if we implement the prefix icon on subtitle we can then use `list-item` in toolbar large header too.
what do you think? @flexsurfer @errorists

File: src/status_im/ui/components/list_item/views.cljs
Comment (tbenr) [reply src/status_im/ui/components/list_item/views.cljs]: sure, i miss that :)

**Comment by bitsikka** [comment]
> @bitsikka thanks for sharing!
> 
> from https://www.figma.com/file/cb4p8AxLtTF3q1L6JYDnKN15/Index?node-id=794%3A173
> 
> image
> 
> if we implement the prefix icon on subtitle we can then use `list-item` in toolbar large header too.
> what do you think? @flexsurfer @errorists

😄 oh boy, similar in structure as they are in some cases (toolbar and list-item), I think the `list-item` is doing quite a bit already, and meant only to be "_list_ item". I think it is better to keep `toolbar` family of components separate. Also better to not overload semantics.

I do however would like for `toolbar` component(s) to be re-structured (which we have left as they have been so far), such that it does not use so much `:position :absolute` to serve its purpose 😛.

**Comment by flexsurfer** [comment]
@tbenr fixed list-item here https://github.com/status-im/status-react/pull/8824

**Comment by flexsurfer** [comment]
as you can see, we already have edge cases, so i wouldn't add more in list-item

**Comment by tbenr** [comment]
@flexsurfer oky, waiting for the merge so i can rebase and integrate

**Comment by bitsikka** [comment]
> @flexsurfer oky, waiting for the merge so i can rebase and integrate

> @tbenr fixed list-item here #8824

@flexsurfer @tbenr 

if you can wait a little bit I am pushing #8070 in just a little bit, in which among other additions, includes this fix as well 🙏

**Comment by tbenr** [comment]
@bitsikka no problem for me! the first will be merged the first i'll integrate :)

**Comment by bitsikka** [comment]
> @bitsikka no problem for me! the first will be merged the first i'll integrate :)

sry :) no need to wait. For one thing, it is not really up to me. For another, while I did push a reviewable PR for #8070 it will probably take a while before it is ready to be merged. Waiting on code/design review.

File: src/status_im/contact/core.cljs
Comment (yenda) [reply src/status_im/contact/core.cljs]: for sure we don't, the question is do we have a "shhext_deleteContact" rpc method to replace it? @cammellos 

if we don't I'd say just add a TODO here that says the contact is not actually remove, because this is a UI PR and there is no point blocking it waiting for a core feature. Remove that :data-store/tx it the delete-contact-tx function won't exist once you rebase anyway.

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (yenda) [inline src/status_im/ui/screens/profile/contact/views.cljs:20]: not sure if you use emacs call cljr-refactor clean-ns on that namespace to remove unused deps and order them alphabetically, there should be a similar feature in cursive

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (yenda) [inline src/status_im/ui/screens/profile/contact/views.cljs]: typo

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (yenda) [reply src/status_im/ui/screens/profile/contact/views.cljs]: contact will always be a map, wallet and accounts will be part of it. I'd just remove this fn a use contact directly

File: src/status_im/ui/components/large_toolbar/view.cljs
Comment (yenda) [inline src/status_im/ui/components/large_toolbar/view.cljs]: put that in the docstring between fn symbol and arguments

File: src/status_im/ui/screens/profile/components/sheets.cljs
Comment (yenda) [inline src/status_im/ui/screens/profile/components/sheets.cljs]: don't refer even these macros, use the `views/` ns alias. I known some ns don't do it but that's old code or mistakes

**Review by yenda — COMMENTED** [review]
kudos and thanks for all your contributions on the project!

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs:20]: I actually use vscode with https://marketplace.visualstudio.com/items?itemName=betterthantomorrow.calva
need to find something similar..

Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs:20]: @yenda cant find anything.. do you know some command line plugin ? i use `lein cljfmt fix <>` to do final formatting but doesn't fix namespaces...

Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs:20]: i manually removed unused but not reordered...

File: src/status_im/contact/core.cljs
Comment (tbenr) [reply src/status_im/contact/core.cljs]: i added a TODO

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs]: @yenda I missed this comment, later this afternoon i'll remove it.

**Comment by bitsikka** [comment]
@tbenr all done with `list-item` 👍 

I have not tested this PR in a device, but only looked at the code. The alternate implementation/technique of large-header you did in this PR is very inspirational for use in variety of other screens/components seen in other areas :) 👍 

same technique is probably quite useful here:
- home screen chat search which has bugs/TODOs atm
- https://github.com/status-im/status-react/issues/8447#issuecomment-515402346

**Comment by tbenr** [comment]
@bitsikka great to hear you! amazing job with #8070! Glad to know my work here has been useful :)
I'll rebase here this evening!

**Comment by tbenr** [comment]
btw #8447 goes in the direction of my first experiment where the toolbar was a single component dynamically resizing and animating content. It requires some dirty absolute positions and padding but was working...

**Comment by flexsurfer** [comment]
@tbenr could you resolve conflicts, please

**Comment by tbenr** [comment]
@flexsurfer not that easy this time, working on it..

**Comment by tbenr** [comment]
@flexsurfer i rebased but i see we have build errors, locally i can build release-ios, can you have a look to see the reason?

**Comment by flexsurfer** [comment]
@tbenr something wrong with tests, they work fine in develop, could you try to run `lein test-cljs` locally for this branch?

**Comment by tbenr** [comment]
@flexsurfer should work now

File: src/status_im/contact/core.cljs
Comment (flexsurfer) [inline src/status_im/contact/core.cljs]: @cammellos could you assists please?

File: src/status_im/ui/components/list/views.cljs
Comment (flexsurfer) [inline src/status_im/ui/components/list/views.cljs:227]: why don't you want just to add animated to props?

Comment (tbenr) [reply src/status_im/ui/components/list/views.cljs:227]: @flexsurfer the idea is to have another map which is not passed to the chosen react class. maybe `:animated?` doesn't effect the behaviour but in general it is not always the case...

File: src/status_im/ui/components/large_toolbar/view.cljs
Comment (yenda) [inline src/status_im/ui/components/large_toolbar/view.cljs:65]: the docstring is before the arguments, if you put it after it is evaluated, because it is a string and there is code afterward it is not noticeable, but that is not considered a docstring and won't be picked up by dev tools for instance.

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (yenda) [inline src/status_im/ui/screens/profile/contact/views.cljs:95]: this should be done in the subscription

Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs:95]: @yenda does it apply here https://github.com/status-im/status-react/blob/09c8d858ed087dc6402f989559d8b17fe0f7cd8a/src/status_im/ui/screens/profile/user/views.cljs#L79 as well?

Comment (yenda) [reply src/status_im/ui/screens/profile/contact/views.cljs:95]: yes we try to avoid doing any computations in the views

Comment (tbenr) [reply src/status_im/ui/screens/profile/contact/views.cljs:95]: @yenda any suggestion on how put it in subscription?

i tryed 

```
  (views/letsubs [list-ref (reagent/atom nil)
                  contact  [:contacts/current-contact]
                  displayed-name       (multiaccounts/displayed-name contact)]
```

but i get "unknown".

haven't found a good example in the code...

File: src/status_im/ui/components/large_toolbar/view.cljs
Comment (tbenr) [reply src/status_im/ui/components/large_toolbar/view.cljs:65]: oops:)

File: src/status_im/contact/core.cljs
Comment (cammellos) [reply src/status_im/contact/core.cljs]: should we remove the contact completely (needs a new endpoint in status-go), or is it enough to remove the `contacts/added` tag (does not need a new endpoint)? 

If we delete it completely, it will come back (maybe not in the list if we are filtering by `contact/added` tag), next contact update, so not sure how useful it is, but I don't have much context on this task.

Comment (yenda) [reply src/status_im/contact/core.cljs]: I suppose we just want to remove contacts/added, it removes the user from the contact list right?

File: src/status_im/ui/screens/profile/contact/views.cljs
Comment (yenda) [reply src/status_im/ui/screens/profile/contact/views.cljs:95]: Leave it as it was for now it's fine

File: src/status_im/contact/core.cljs
Comment (cammellos) [reply src/status_im/contact/core.cljs]: Not sure about that, I would guess so as it should only show contacts that we added.

Essentially we can just remove the `contact/added` tag and call `save-contact` in `status-im.data-store.contacts`

Comment (tbenr) [reply src/status_im/contact/core.cljs]: @cammellos @yenda take a look to the latest push

**Review by flexsurfer — APPROVED** [review]

File: src/status_im/contact/core.cljs
Comment (yenda) [inline src/status_im/contact/core.cljs]: I'm not sure you should dissoc here but rather update the contact with new contact as well

Comment (tbenr) [reply src/status_im/contact/core.cljs]: right..

Comment (tbenr) [reply src/status_im/contact/core.cljs]: @yenda fixed

**Review by yenda — APPROVED** [review]

**Comment by yenda** [comment]
@tbenr seems like there is some runtime error according to e2e tests

**Comment by tbenr** [comment]
hum.. something related to Animated.FlatList for sure... i'll try to run on my physical android

**Comment by tbenr** [comment]
@yenda on my physical android 8 works ok.. i missed de fiddle js_dependencies

**Comment by tbenr** [comment]
do you know which android version is running the e2e emulator?

**Comment by flexsurfer** [comment]
@Serhy do you know?

**Comment by tbenr** [comment]
sorry I inadvertently closed the PR.. 👎

**Comment by Serhy** [comment]
E2e emulators are running on Android 8.0
I've installed https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190905-115808-833a55-pr.apk build on real Samsung J7 with Android 8.1 and have the same crash once navigation to `Chats` main screen happens.
So its:
1) Install https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190905-115808-833a55-pr.apk 
2) Open status.
3) Create new account

Screenshot 2019-09-05 at 16 44 22


Same error happens when recover seed phrase or login to account

**Comment by tbenr** [comment]
thanks @Serhy  ok i replicated.. probably it happens only in release build.. maybe is the extern.js let me do some tests.

File: components/src/status_im/ui/components/react.cljs
Comment (yenda) [inline components/src/status_im/ui/components/react.cljs:84]: you need to add FlatList to extern.js

Comment (tbenr) [reply components/src/status_im/ui/components/react.cljs:84]: i was suspecting :)

**Comment by tbenr** [comment]
@flexsurfer @Serhy we can try again

**Comment by flexsurfer** [comment]
@tbenr could you resolve conflicts please, action_button has been removed, use [list-item {:theme :action}] instead

**Comment by tbenr** [comment]
@flexsurfer ok, i'll do it later this afternoon

**Comment by tbenr** [comment]
@flexsurfer done

**Comment by Serhy** [comment]
@tbenr everything looks good (tested IOS 12, Android builds), few points to address:
Screenshot 2019-09-10 at 11 41 07

1) `Block this user`(`Unblock this user`) should be 16px from left device border (aligned to `Profile details` label above)
2) No shortened user public key below username, but we do have on mockup. 
3) Share icon color should be ```/* 40% Dark Grey * 
background: rgba(147, 155, 161, 0.4);```

**Comment by tbenr** [comment]
@Serhy ok almost fixed.. i'll push it this evening

**Comment by tbenr** [comment]
@Serhy ready to test again

**Comment by churik** [comment]
@tbenr 
### 1. Profile details element has animation  on tap, but no action is provided
I would prefer to see popup with chat key is opened that if I tap below "Profile details" on user picture or username. Currently, popup is shown only if I tap on icon "Share"
screencast 2019-09-12 16-59-41 2

Great work!
Meanwhile, I'll take a look into e2e (should be fixed accordingly)

### 2. Restore accessibility-ids
2.1 for "Block this user" before changes had 'block-contact' accessibility_id
2.2 for chat-key - 'profile-public-key'

### 3. question to @flexsurfer: prompts are introduced, but currently there is no place where I can see wallet address of another user and check the validity of these prompts. 

Should it be implemented separately, right? if there is no issue about it, it should be logged.
image

**Comment by errorists** [comment]
re: 1st of Chu’s comments, that is correct, the entire list item should be tappable and followed by popup, not only the icon.

**Comment by tbenr** [comment]
@errorists @churik yes makes sense. done. pushing!

**Comment by churik** [comment]
@tbenr please take a look also to 2 in https://github.com/status-im/status-react/pull/8776#issuecomment-530878636 (updated comment)

**Comment by tbenr** [comment]
> @tbenr please take a look also to 2 in [#8776 (comment)](https://github.com/status-im/status-react/pull/8776#issuecomment-530878636)

not 100% clear to me what you intend here... you mean you want to display address to be added\removed?

**Comment by churik** [comment]
@tbenr 

>  2. Restore accessibility-ids
2.1 for "Block this user" before changes had 'block-contact' accessibility_id
2.2 for chat-key - 'profile-public-key'

**Comment by tbenr** [comment]
lol, i'm blind.

**Comment by tbenr** [comment]
@churik regarding 2.2 i added the accessibility label on the list item (the one you click to get the qrcode popup). I hope it is what you intented. If yes, 1. and 2. are fixed.

**Comment by flexsurfer** [comment]
@churik yes sorry, we do not share wallet address, @tbenr sorry could you comment text in add/remove contact bottom sheets for now, and keep only buttons, we'll return them later, thanks!

**Comment by errorists** [comment]
@flexsurfer Actually, do we need the bottom sheet at all then? Since we don't share wallet addresses, it doesn't serve a purpose anymore

**Comment by flexsurfer** [comment]
@errorists right we don't need them for now, but better to keep them in the code @tbenr . just move event from sheet to profile items thanks

**Comment by tbenr** [comment]
@flexsurfer ok, can't do it before this evening, i'm not at home.

**Comment by churik** [comment]
@tbenr 
Also please if you don't mind and have a time, add accessibility-id to all labels in user profile (to 3-random name, avatar, send message, add to contacts (remove from contacts), user profile details).
It will help e2e tests a lot!
Thank you for your patience :)

**Comment by tbenr** [comment]
@churik sure :)

**Comment by tbenr** [comment]
@flexsurfer @churik done (hopefully I covered all labels)
i rebase resolving conflicts and i did a cleanup removing unused styles and functions.

**Comment by flexsurfer** [comment]
@tbenr could you check cljs tests, they fail

**Comment by tbenr** [comment]
@flexsurfer something weird in `en.json`, everything I add in there brakes test.
is something changed in i18n stuff which i have to take in account?

**Comment by tbenr** [comment]
It happens in develop too.
I checkout develop
make test -> OK
I add an entry in en.json
make clean
make test -> FAIL (same error i get in this PR)

**Comment by tbenr** [comment]
I found the issue, i was coping paste json fields from macos Note app. this was introducing a dirty\invisible 2byte utf-8 char... damn.

**Comment by tbenr** [comment]
@flexsurfer tried to rebase again but jenkins build isn't triggered..

**Comment by yenda** [comment]
@jakubgs  another jenkins build not triggering

**Comment by jakubgs** [comment]
Sorry for that, this time was about permissions due to branch not being in the repo.
Should be building now.

**Comment by tbenr** [comment]
@flexsurfer @churik @Serhy can we try to test again?

**Comment by rachelhamlin** [comment]
@tbenr apologies for any delay - we're on our team offsite in Istanbul this week so will be responding a bit more slowly than usual. Would love to get this one finalized so we can pay you! Will see if we can prioritize it.

**Comment by bitsikka** [comment]
Merging this will also help me with PRs I'm working on 🙌 :)
Thanks! @tbenr and everyone

**Comment by churik** [comment]
@tbenr sorry for delay, we all on offsite in Istanbul.
I'll come back it today or tomorrow.

**Comment by churik** [comment]
@tbenr two small issues:
1. for label "Unblock this user" we have accesibility-id "block-contact" and vice versa, so they are messed up.
2. for "remove from contacts" translation is missing
image

Thank you for hard work!

**Comment by tbenr** [comment]
@churik damn :) i'll fix it when back home... thanks for patient

**Comment by tbenr** [comment]
@churik *MAYBE* ***HOPEFULLY*** this is the last time :)

**Comment by churik** [comment]
@tbenr :)
we lost dialog about the confirmation of "Block contact".
This is last one from my side.
https://github.com/status-im/status-react/pull/8776#issuecomment-524020861 - bottom sheet is required for blocking contact. I'm really sorry for confusion - but afaik @flexsurfer meant only for add/remove contact - so bottom sheet with explanation should definitely be shown when you blocked contact.

And I believe we already got confirmation from @errorists that he is OK with new view.
So only get back message when blocking users.
And specific thank you for accessibility-ids, it helped a lot!

**Comment by tbenr** [comment]
@churik ahh ok I reactivate it once back home. No problems:)

**Comment by tbenr** [comment]
@churik reenabled sheet on block, rebased too.

**Comment by churik** [comment]
@tbenr many thanks for your patience!

**Pull request merged**
